inc/packages/namespace.php

```diff
@@ -703,8 +705,20 @@ function delete_cached_did_for_install(): void {
 function maybe_rename_source_selection( string $source, string $remote_source, WP_Upgrader $upgrader ) {
 	global $wp_filesystem;
 
+	$type = $upgrader instanceof Plugin_Upgrader ? 'plugin' : ( $upgrader instanceof Theme_Upgrader ? 'theme' : '' );
 	$did = get_site_transient( CACHE_DID_FOR_INSTALL );
 
+	if ( ! $did ) {
+		if ( empty( $type ) ) {
+			return $source;
+		}
+		$did_doc = get_did_by_path( $source, $type );
+		if ( is_wp_error( $did_doc ) ) {
+			return $did_doc;
+		}
+		$did = $did_doc->get_id();
+	}
+
 	if ( ! $did ) {
```

Comment:
@afragen I'm not sure this condition can ever be true now. Either we bail on `empty( $type )` or `is_wp_error( $did_doc )`, but otherwise, `$did === $did_doc->get_id()`.

Any idea if there's some way that this condition can ever be true?

Reply:
I don't think this will ever be true now